src/App.tsx

```diff
@@ -15,47 +15,140 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  let visibleGoods = [...goods];
+
+  switch (sortType) {
+    case SortType.ALPABET:
+      visibleGoods = visibleGoods.sort((a, b) => a.localeCompare(b));
+      break;
+
+    case SortType.LENGTH:
+      visibleGoods = visibleGoods.sort((a, b) => a.length - b.length);
+      break;
+
+    default:
+      //
+  }
+
+  if (isReversed) {
+    return visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [appState, setAppState] = React.useState({
+    sortType: SortType.NONE,
+    isReversed: false,
+  });
+
+  const handelSortAlphbet = () => {
+    setAppState(prevState => {
+      return {
+        ...prevState,
+        sortType: SortType.ALPABET,
+      };
+    });
+  };
+
+  const handelSortLength = () => {
+    setAppState(prevState => {
+      return {
+        ...prevState,
+        sortType: SortType.LENGTH,
+      };
+    });
+  };
+
+  const handelReverse = () => {
+    setAppState((prevState) => {
+      return {
+        ...prevState,
+        isReversed: !prevState.isReversed,
+      };
+    });
+  };
+
+  const handelReset = () => {
+    setAppState({
+      isReversed: false,
+      sortType: SortType.NONE,
+    });
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={appState.sortType === SortType.ALPABET
+            ? 'button is-info'
+            : 'button is-info is-light'}
+          onClick={handelSortAlphbet}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={appState.sortType === SortType.LENGTH
+            ? 'button is-success'
+            : 'button is-success is-light'}
+          onClick={handelSortLength}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={appState.isReversed
+            ? 'button is-warning'
+            : 'button is-warning is-light'}
+          onClick={handelReverse}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(appState.isReversed || appState.sortType !== SortType.NONE) && (
```

Comment:
such long conditions are better to put in a varibale and use just one variable here